<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <include file="$(find gazebo_ros)/launch/empty_world.launch">
    <arg name="world_name" value="$(find visual_path_husky)/worlds/path_floor.world"/>
  </include>

  <arg name="laser_enabled" default="true" doc="Enable husky laser"/>
  <arg name="kinect_enabled" default="true" doc="Enable husky kinect"/>
  <arg name="robot_namespace" default="/"/>

  <include file="$(find husky_gazebo)/launch/spawn_husky.launch">
    <arg name="laser_enabled" value="$(arg laser_enabled)"/>
    <arg name="kinect_enabled" value="$(arg kinect_enabled)"/>
    <arg name="robot_namespace" value="$(arg robot_namespace)"/>
    <arg name="x" value="-6.0"/>
    <arg name="y" value="1.2"/>
    <arg name="z" value="0.02"/>
    <arg name="yaw" value="0.0"/>
  </include>

  <node name="safety_controller" pkg="visual_path_husky" type="safety_controller.py" output="screen" />
  <node name="line_following_node" pkg="visual_path_husky" type="simple_follow_line.py" output="screen" />
  <node name="circle_performer" pkg="visual_path_husky" type="circle_plus_marker.py" output="screen" />

</launch>
